Add a check for AAAA records

seveas opened the pull request:
GitHub pages only supports ipv4 at this time. Having AAAA records for a
pages domain is wrong and will lead to certificate enrollment failures,
as Let's Encrypt would connect to the wrong host.

parkr commented on lib/github-pages-health-check/errors/invalid_aaaa_record_error.rb:
```diff
@@ -0,0 +1,19 @@
+        def message
+          <<-MSG
+             Your site's DNS settings are using a custom subdomain, #{domain.host},
+             that's set up with an AAAA record. GitHub pages currently does not support
```
`GitHub Pages`

parkr commented on lib/github-pages-health-check/errors/invalid_aaaa_record_error.rb:
```diff
@@ -0,0 +1,19 @@
+          <<-MSG
+             Your site's DNS settings are using a custom subdomain, #{domain.host},
+             that's set up with an AAAA record. GitHub pages currently does not support
+             ipv6 and this AAAA record will interfere with https certificate creation.
```
`IPv6`, and `HTTPS`

Not sure if we should say anything about HTTPS directly (though it applies!) because having IPv6 records could actually cause issues serving for clients that lookup IPv6 first before IPv4. So it could be a bigger issue.

parkr commented on spec/github_pages_health_check/domain_spec.rb:
```diff
@@ -151,6 +154,16 @@
+        expect(subject).to be_an_a_record
+        expect(subject).to be_a_valid_domain
+        expect(subject.invalid_aaaa_record?).to be_truthy
+      end
```
Can we also add a test to ensure it raises the correct error?

```ruby
it "raises InvalidAAAARecordError" do 
  expect(-> { subject.check! }).to raise_error(Errors::InvalidAAARecordError)
end
```

This ensures that the checking will surface this error in the right order in the `check!` method.

parkr commented on lib/github-pages-health-check/domain.rb:
```diff
@@ -286,6 +293,11 @@ def a_record?
 
+      def aaaa_record?
+        return unless dns?
+        dns.index { |r| r.type == Dnsruby::Types::AAAA }
```
Would `dns.any?` make better sense here? It returns a boolean value which is what I think you want here.

seveas commented:
Thanks @parkr, I force-pushed an update that addresses your feedback (and renames `aaaa_record?` to `aaaa_record_present?` to be consistent with `mx_record_present?`)

seveas commented:
Y'know what would be useful: running the tests *after* saving the file in your editor 🤦‍♂️ 

And that actually found a couple of bugs.

parkr commented:
Do you any further changes? If not, I'll push a 1.9.0 😄

seveas commented:
Not sure, depends on what other issues come up while looking at support tickets.

parkr commented:
Cool, cut a release: https://github.com/github/pages-health-check/releases/tag/v1.9.0
